### ISSUE TITLE
Investigate Environment.TickCount64 potential issues

### ISSUE THREAD

**Issue by sebastienros** [dotnet/runtime#52407]
Since this bug https://github.com/dotnet/runtime/issues/51935 was triggered by `Environment.TickCount64` that can wrap around, I checked some usages of the property and found potential similar issues. There is a common pattern similar to this code:

https://github.com/dotnet/runtime/blob/33db0a283546c56f49b5af7148a6ac494a61efa5/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs#L1982-L1988

Am I right that in this case `now` could become negative, and only reach the expected value when the cycle is reached again?

This might not be a critical issue in this case but I assume some code checks should be done to ensure that some code would not be blocked in a specific state if the machine is up for more than 25 days.

**Comment by jkotas** [comment]
> Since this bug #51935 was triggered by Environment.TickCount64

This bug was triggered by `Environment.TickCount` that returns 32-bit number, not 64-bit number.

`Environment.TickCount64` cannot really wrap around in practice.

**Comment by KevinCathcart** [comment]
`TickCount64` takes `292 277 266 years` to wrap around, while the 32 bit one `TickCount` takes `49.7 days`.

The theadpool code likely used the the 32 bit one because it was designed long ago, for 32 bit hardware, where using 64 bit would be slower, and if they were doing unsigned arithmetic everything was fine. But when the Portable version of the thread code was created they did not realize the issue with using signed arithmetic. Indeed, the code almost worked, and it would only break if an app got hibernated for over 3 weeks, or in the initialization scenario.

Furthermore, the thread pool code uses `Volatile.Read`/`Volatile.Write` to read the data, and for a 64 bit value that requires a slower operation on 32 bit machines to ensure an atomic read. (It then goes on to make use of the acquire/release semantics to read the other value without any synchronization privatives, which is would only safe for a 32 bit value on a 32 bit machine.)

Basically using the 64 bit count there would meaningfully impact performance on 32 bit machines.
